build.sbt

```diff
@@ -208,7 +208,7 @@ TaskKey[Unit](
   // would be good to be able to do this without exiting the JVM...
   val sv = scalaVersion.value
   val cmd =
-    if (sys.env.contains("APPVEYOR")) """C:\sbt\bin\sbt.bat""" else "sbt"
+    if (sys.env.contains("APPVEYOR")) """C:\sbt\bin\sbt.bat""" else "../../sbt"
```

Comment:
This looks specific to your machine

Reply:
That's actually to make it works in Travis. The line fetching sbt with curl 
```
curl https://raw.githubusercontent.com/paulp/sbt-extras/master/sbt -o sbt ; chmod 755 sbt
```

Will not work when executing the prewarm task, since sbt is two levels up, while the task is running under `testing/cahe`. Reverting back to `sbt` only gives error file not found. 

I am not sure to fix this here, or in .travis.yml, (by for example setting current path to be able to access sbt)

Reply:
About the too big pull request, I'll come back to this after work or tomorrow, and make smaller PR just with the sensitivity change.

Reply:
aaah, I thought this was teh appveyor file. ok, that makes sense. But I think you need a `if (appveyor) ... else if (travis) ... else ...` or it'll break things for everybody else.

Reply:
I will just close this one and make a smaller pull request just the sensitivity modifier. The refactoring of FileWatchService is a bit too big for me also.